application_boilerplate/js/MapUrlParams.js

```diff
@@ -0,0 +1,290 @@
+        deferred.reject();
+      }
+
+      return deferred.promise;
+    },
+    _createCenter: function(){
+    //?center=-13044705.25,4036227.41,102113&level=12 or ?center=-13044705.25;4036227.41;102113&level=12
+    //?center=-117.1825,34.0552&level=12 or ?center=-117.1825;34.0552&level=12
+    var deferred = new Deferred();
+    var centerArray = this._splitArray(this.center);
+    if (centerArray.length === 2 || centerArray.length === 3) {
+      var x = parseFloat(centerArray[0]);
+      var y = parseFloat(centerArray[1]);
+      if (isNaN(x) || isNaN(y)) {
+        x = parseFloat(centerArray[0]);
+        y = parseFloat(centerArray[1]);
+      }
+      if (!isNaN(x) && !isNaN(y)) {
+        var wkid = 4326;
+        if (centerArray.length === 3 && !isNaN(centerArray[2])){
+          wkid = parseInt(centerArray[2], 10);
+        }
+
+        var point = new Point(x, y, new SpatialReference(wkid));
+        this._project(point).then(lang.hitch(this, function(pPoint){
+          var projectedResult = (pPoint) ? {"center": pPoint} : null;
+          deferred.resolve(projectedResult);
+        }),deferred.reject);
+
+      }else{
+        deferred.reject();
+      }
+    }else{
+      deferred.reject();
+    }
+
+      return deferred.promise;
+    },
+    _createLevel: function(){
+      // ?level=4
+      var deferred = new Deferred();
+      var level = this.level;
+      if (level) {
+        deferred.resolve({"zoom":parseInt(level)});
+      }else{
+        deferred.reject();
+      }
+      return deferred.promise;
+    },
+    _createMarker: function(){
+      var deferred = new Deferred();
+      /*
+      ?marker=-117;34;4326;My%20Title;http%3A//www.daisysacres.com/images/daisy_icon.gif;My%20location&level=10
+      ?marker=-117,34,4326,My%20Title,http%3A//www.daisysacres.com/images/daisy_icon.gif,My%20location&level=10
+      ?marker=-13044705.25,4036227.41,102100,My%20Title,http%3A//www.daisysacres.com/images/daisy_icon.gif,My%20location&level=10
+      ?marker=-117,34,,My%20Title,http%3A//www.daisysacres.com/images/daisy_icon.gif,My%20location&level=10
+      ?marker=-117,34,,,,My%20location&level=10
+      ?marker=-117,34&level=10
+      ?marker=10406557.402,6590748.134,2526&level=10
+      */
+      require(["esri/symbols/PictureMarkerSymbol", "esri/graphic","esri/dijit/PopupTemplate"], lang.hitch(this, function(PictureMarkerSymbol, Graphic, PopupTemplate){
+         var markerArray = this._splitArray(this.marker);
+         if (markerArray.length >= 2 &&
+         !isNaN(markerArray[0]) &&
+         !isNaN(markerArray[1])) {
+            var x = parseFloat(markerArray[0]),
+              y = parseFloat(markerArray[1]),
+              wkid = markerArray[2] || null,
+              description = markerArray[3] || null,
+              icon_url = markerArray[4] || this.defaultMarkerSymbol || null,
+              label = markerArray[5] || null;
+
+
+              var markerSymbol = new PictureMarkerSymbol(icon_url, 26, 26);
```

Comment:
I think in the URL, we can just have the symbol URL, but via the config, it would be nice to have the width and height options. If you define a new symbol, its too hard to set the size of it currently.